src/hooks/use-markdown-processor.tsx

```diff
@@ -244,8 +246,10 @@ const CodeBlock = ({ children, className }: JSX.IntrinsicElements["code"]) => {
             aria-label="copy code to clipboard"
             title="Copy code to clipboard"
             onClick={() => {
-              navigator.clipboard.writeText(children?.toString() ?? "");
-              setCopied(true);
+              if (ref.current) {
+                navigator.clipboard.writeText(ref.current.innerText.replace(/\\n/g," ") ?? "");
```

Comment:
Is the `replace` necessary? I would think you would want to include new lines when copying code?

Reply:
Right; it doesn't make sense to keep it in this context. Fixed! 👍🏼